web/src/settings_users.ts

```diff
@@ -191,29 +191,56 @@ function role_selected_handler(
     widget.render();
 }
 
-function get_roles(): {
-    unique_id: number;
-    name: string;
-}[] {
+function count_users_by_role(user_ids: number[]): Record<number, number> {
+    const role_counts: Record<number, number> = {};
+
+    for (const user_id of user_ids) {
+        const user = people.get_by_user_id(user_id);
+        const role_code = user.role;
+
+        role_counts[role_code] = (role_counts[role_code] ?? 0) + 1;
+    }
+
+    return role_counts;
+}
+
+function get_roles_with_counts(user_ids: number[]): {unique_id: number; name: string}[] {
+    const role_counts = count_users_by_role(user_ids);
     return [
-        {unique_id: 0, name: $t({defaultMessage: "All roles"})},
+        {
+            unique_id: 0,
+            name: $t(
+                {defaultMessage: "{all_roles} ({count})"},
+                {
+                    all_roles: $t({defaultMessage: "All roles"}),
+                    count: user_ids.length,
+                },
+            ),
+        },
         ...Object.values(settings_config.user_role_values)
             .map((user_role_value) => ({
                 unique_id: user_role_value.code,
-                name: user_role_value.description,
+                name: $t(
+                    {defaultMessage: "{description} ({count})"},
```

Comment:
`user_role_value.description` is already translated, so I am not sure whether it will be fine to tag already translated text for translation.

Reply:
@sahil839, I followed your suggestion as discussed [here](https://github.com/zulip/zulip/pull/32147#discussion_r1838225857). Could you please let me know if I interpreted it correctly, or if there are any adjustments needed? Thank you!

Reply:
I think that would work based on discussion [here](https://github.com/zulip/zulip/pull/26766/files#r1328971956), but I am not sure if we would prefer some other way.